kernel/src/main/scala-2.13+/cats/kernel/compat/scalaVersionSpecific.scala

```diff
@@ -35,4 +37,8 @@ private[cats] object scalaVersionSpecific {
   implicit class iterableOnceExtension[A](private val io: IterableOnce[A]) extends AnyVal {
     def reduceOption(f: (A, A) => A): Option[A] = io.iterator.reduceOption(f)
   }
+
+  implicit class setExtension[A](private val s: java.util.Set[A]) extends AnyVal {
+    def asScala: mutable.Set[A] = SetHasAsScala(s).asScala
```

Comment:
I doubt that `SetHasAsScala` is a part of any API. Not mention this call can allocate a class instance without a reason. I'd probably give a shot to `scala.jdk.javaapi.CollectionConverters` here, e.g.:
```scala
import scala.jdk.javaapi.CollectionConverters

...

def asScala: Iterable[A] = CollectionConverters.asScala(x)

```